configure.ac

```diff
@@ -0,0 +1,191 @@
+# whose definition is guarded by _GNU_SOURCE
+AC_USE_SYSTEM_EXTENSIONS
+AX_CFLAGS_WARN_ALL
+AC_PROG_INSTALL
+
+# Process options
+
+AC_ARG_ENABLE([systemd],
+  [AS_HELP_STRING([--enable-systemd],
+    [enable systemd support @<:@default=check@:>@])],,
+  [enable_systemd=check])
+
+AC_ARG_ENABLE([selinux],
+  [AS_HELP_STRING([--enable-selinux],
+    [enable selinux support @<:@default=check@:>@])],,
+  [enable_selinux=check])
+
+# Checks for libraries
+
+# Unclear why pthread is needed.
+# Uncomment and replace ? by called procedure requiring -lpthread.
+#AC_SEARCH_LIBS([?], [pthread])
+
+AC_SEARCH_LIBS([uncompress], [z])
+
+# Enable systemd per option by setting HAVE_SYSTEMD
+
+AS_CASE([${enable_systemd}],
+  [yes|check],
+    [AC_MSG_CHECKING([systemd])
+     AC_SEARCH_LIBS([sd_listen_fds], [systemd-daemon])
+     AC_CHECK_FUNC([sd_listen_fds],
+       [AC_MSG_RESULT([enabled: systemd])
+        AC_DEFINE([HAVE_SYSTEMD],[1],
+          [Define to 1 to enable systemd support])],
+       AS_CASE([${enable_systemd}],
+         [yes],
+           AC_MSG_ERROR([systemd support not available]),
+	 AC_MSG_RESULT([disabled: systemd])))],
+  [no],,
+  [AC_MSG_ERROR([invalid argument, --enable-systemd=${enable_systemd}])])
+
+# Enable selinux per option by setting HAVE_SETCON
+
+AS_CASE([${enable_selinux}],
+  [yes|check],
+    [AC_SEARCH_LIBS([setcon], [selinux])
+     AC_CHECK_FUNCS([setcon],
+       AC_MSG_RESULT([enabled: selinux]),
+       AS_CASE([${enable_selinux}],
+         [yes],
+           AC_MSG_ERROR([selinux support not available]),
+         [check],
+           AC_MSG_RESULT([disabled: selinux])))],
+  [no],
+    AC_MSG_RESULT([disabled: selinux]),
+  [AC_MSG_ERROR([invalid argument, --enable-selinux=${enable_selinux}])])
+
+# The check library is used for the unit testing.  Not bothering to
+# implement an --enable-tests option, as they are only run by "make
+# check" so there is no user-level difference.  If the unit tests
+# won't even *compile* there is probably a better solution than just
+# turning them off, and build time is minimal.
+
+PKG_CHECK_MODULES(CHECK,[check],
+  AC_MSG_RESULT([enabled: unit testing]),
+  AC_MSG_RESULT([disabled: unit testing]))
+
+AM_CONDITIONAL([HAVE_CHECK],[test -z "${CHECK_PKG_ERRORS}"])
+
+# For each commented AC_SEARCH_LIBS below, uncomment and replace ? by
+# called procedure provided only by that library on the platform(s) in
+# question.
+
+# for SunOS / solaris:
+#AC_SEARCH_LIBS([?], [nsl])
+
+# for BeOS:
+#AC_SEARCH_LIBS([?], [bind])
+#AC_SEARCH_LIBS([?], [bsd])
+
+# for SunOS / solaris / BeOS:
+#AC_SEARCH_LIBS([?], [socket])
+
+# for Haiku:
+#AC_SEARCH_LIBS([?], [network])
+
+# for MS windows32:
+#AC_SEARCH_LIBS([?], [ws2_32])
+#AC_SEARCH_LIBS([?], [iphlpapi])
+
+# Checks for headers
+
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h netdb.h netinet/in.h stdint.h stdlib.h string.h sys/ioctl.h sys/param.h sys/socket.h sys/time.h syslog.h termios.h unistd.h systemd/sd-daemon.h selinux/selinux.h])
```

Comment:
stdint.h stdlib.h string.h unistd.h are standard C headers (even since C89).
all the other headers apart from sys/ioctl.h, sys/param.h and the systemd and selinux headers are standard POSIX headers. so i see little point in testing for their existence.